src/yaml/yamlUtils.ts

```diff
@@ -0,0 +1,257 @@
+    startPositionForInsertion: The starting Position for the insert range applied to the generated code completions
+    endPositionForInsertion: The ending Position for the insert range applied to the generated code completions
+    shouldPrependSpace: A boolean indicating whether the generated code completions should have a space prepended to them
+*/
+export function processYamlDocForCompletion(document: TextDocument, position: Position): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    const lineOffsets = getLineOffsets(document.getText())
+    const { currentLine, currentLineEnd } = getCurrentLine(document, position, lineOffsets)
+
+    if (currentLine.indexOf(':') === -1) {
+        return processLineWithoutColon(document, position, currentLine, currentLineEnd)
+    } else {
+        return processLineWithColon(document, position, currentLine, currentLineEnd)
+    }
+}
+
+function processLineWithoutColon(document: TextDocument, cursorPosition: Position, currentLine: string, currentLineEnd: number): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    let modifiedDocText: string
+    let shouldPrependSpace: boolean = false
+
+    const tempPositionForCompletions: Position = {...cursorPosition}
+    const startPositionForInsertion: Position = {...cursorPosition}
+    const endPositionForInsertion: Position = {...cursorPosition}
+
+    // Since there's no colon to separate the key and value, replace all text after the cursor.
+    endPositionForInsertion.character = currentLine.length
+
+    const lineOffsets = getLineOffsets(document.getText())
+
+    const trimmedLine = currentLine.trim()
+
+    if (trimmedLine.length === 0) {
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                '"":\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+        tempPositionForCompletions.character += 1
+
+    // Trimmed line starts with '-'
+    } else if (trimmedLine.length >= 1 && trimmedLine[0] === '-') {
+        // Set start of insertion range to be immediately after the hyphen.
+        const indexOfHyphen = currentLine.indexOf('-')
+        startPositionForInsertion.character = indexOfHyphen + 1
+        shouldPrependSpace = true
+
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                (!currentLine.endsWith(' ') ? ' ' : '') +
+                ':\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
```

Comment:
we use `document.getText()` and `document.getText().length` in many places. Can we declare a cons at the top to avoid repetition?

Reply:
Updated to define constants for docText and docTextLength